test/gtest/ucp/test_ucp_am.cc

```diff
@@ -570,15 +576,15 @@ class test_ucp_am_nbx : public test_ucp_am_base {
                                          const ucp_am_recv_param_t *rx_param)
     {
 
-        EXPECT_FALSE(m_am_received);
+        EXPECT_LT(m_recv_counter, m_send_counter);
```

Comment:
maybe EXPECT_EQ(0, m_recv_counter) ?

Reply:
That can cause an error because this CB handles a batch of messages sent by the AM header test, the m_recv_counter in this case can be larger than 0.
I think that in the tests that send only one message we can treat 
"EXPECT_LT(m_recv_counter, m_send_counter)" and "EXPECT_EQ(0, m_recv_counter)" the same because m_send_counter is equal to 1.

This explanation is true also for the other common CBs that uses the counters. I can change the EXPECT_LT to EXPECT_EQ only in tests that sending only one message and performing this check in non common code, but again in my perspective if you send only one message both checks can be considered equal.

Reply:
how did it work before this PR then? the test header test that sends multiple messages already existed

Reply:
Different AM_CB's that uses common CB. Only one checked the flags.
Can use it again and check for two different conditions.